### PR TITLE
[configtls] Removed deprecated structs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** <Describe what has changed.>
Removed deprecated structs

**Link to tracking Issue:** <Issue number if applicable>
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/9428
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/9474
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/9548

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
Depends on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31814

**Comment by mx-psi** [comment]
Looks like there are a few fixes missing on contrib:

```
# github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opencensusexporter [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opencensusexporter.test]
Error: ./config_test.go:61:7: unknown field TLSSetting in struct literal of type configtls.ClientConfig
Error: ./factory_test.go:127:7: unknown field TLSSetting in struct literal of type configtls.ClientConfig
Error: ./factory_test.go:141:7: unknown field TLSSetting in struct literal of type configtls.ClientConfig
```

```
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/postgresqlreceiver [github.com/open-telemetry/opentelemetry-collector-contrib/receiver/postgresqlreceiver.test]
Error: ./config_test.go:166:4: unknown field TLSSetting in struct literal of type configtls.ClientConfig
```

File: .chloggen/configtls-remove-deprecated.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (TylerHelmuth) [inline .chloggen/configtls-remove-deprecated.yaml:25]:
```suggestion
change_logs: [api]
```

**Comment by TylerHelmuth** [comment]
@mx-psi those errors are happening because `ClientConfig` and `ServerConfig` use `mapstructure:",squash"` and the type was changed to `Config` from `TLSSetting` - this is a new breaking change being introduced in this PR as part of the removal.  I've created separate changelog entries to help api users see both breaking changes from this PR.  Once this PR is merged I'll update Contrib to use `Config` instead of `TLSSetting` in the tests.

File: config/configtls/configtls.go [hunk 2]
```diff
@@ -75,17 +71,12 @@ type Config struct {
...
-// Deprecated: [v0.96.0] Use ClientConfig instead.
-type TLSClientSetting = ClientConfig
-
 // ClientConfig contains TLS configurations that are specific to client
 // connections in addition to the common configurations. This should be used by
 // components configuring TLS client connections.
 type ClientConfig struct {
 	// squash ensures fields are correctly decoded in embedded struct.
-	TLSSetting `mapstructure:",squash"`
+	Config `mapstructure:",squash"`
```
Comment (astencel-sumo) [inline config/configtls/configtls.go:79]: I'm not convinced that having a field named `Config` in a struct named `ClientConfig` is a good choice of names, but I cannot think of a better name. I don't think `CommonConfig` is any better? This field should ideally not exist at all, but well, Go lang... :roll_eyes:

**Review by astencel-sumo — APPROVED** [review]

File: config/configtls/configtls.go (hunk 2 above)
Comment (TylerHelmuth) [reply config/configtls/configtls.go:79]: I played around with `TLSConfig Config `mapstructure:",squash"`

**Pull request merged**